### ISSUE TITLE
Cannot select build task if no default task is configured

### ISSUE THREAD

**Issue by anatolyburakov** [microsoft/vscode#215520]
Does this issue occur when all extensions are disabled?: Yes, I tried "Extension Bisect" and the issue was reproducible without any extensions enabled

Version: 1.91.0-insider (system setup)
Commit: 0a417782b1781d39803b8f1791cfb22f8835ddae
Date: 2024-06-13T05:48:47.964Z
Electron: 29.4.0
ElectronBuildId: 9593362
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Windows_NT x64 10.0.22631

This is a regression and a duplicate of https://github.com/microsoft/vscode/issues/209953#issuecomment-2140160253 which was closed for some reason.

Steps to Reproduce:

0. Open any directory
1. Create `tasks.json` under `.vscode` directory
2. Create a build task but not set it as default
3. Run `Build` action

Expected result: picker should offer me to select my build task
Actual result: prompt to create a default task

Created task:
```
{
    "version": "2.0.0",
    "tasks": [
        {
            "label": "Test",
            "type": "shell",
            "command": "echo test",
        }
    ]
}
```

This is happening both locally and with Remote SSH functionality.

![image](https://github.com/microsoft/vscode/assets/37181601/63bcdb2d-e9d2-44d3-b22b-c45915eb035d)

**Comment by meganrogge** [comment]
I cannot reproduce the issue - see I have commented out my only `isDefault` task here. This suggests your tasks.json is inaccessible for some reason

<img width="835" alt="Screenshot 2024-06-14 at 9 49 32 AM" src="https://github.com/user-attachments/assets/ad93e5cc-61e7-4b2e-a7ba-1027bcd96d80">

**Comment by anatolyburakov** [comment]
> I cannot reproduce the issue - see I have commented out my only `isDefault` task here.

After a bit of debugging, I think I figured it out. My `tasks.json` did not have tasks of kind `build`, it just had a bunch of tasks that were not assigned to any particular type (see my quoted `tasks.json` - there is no `"group": { "kind": ... }` there). When I set my tasks as "build" group, I can see the list of tasks. If I add `group:kind=build` marker, then I can see the list of tasks when I trigger Build command, if I remove `group`, VSCode says no build tasks found and suggests picking a default.

While this isn't necessarily a regression, this is a change of behavior - before, all tasks were discovered automatically, but now it seems that only tasks of kind "build" are discovered. Was this change intentional?
